### PR TITLE
mgmt, support sdk_automation_typespec

### PR DESCRIPTION
**Pull request by XiaofeiCao**
# Description

1. fix current sdk automation error with arm
2. separate arm and data in sdk_automation_typespec

test-repo-billy: https://github.com/test-repo-billy/azure-rest-api-specs/pull/3293

related issue: https://github.com/Azure/azure-sdk-tools/issues/8040

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: eng/mgmt/automation/generate.py [hunk 1]
```diff
@@ -199,6 +207,116 @@ def sdk_automation_autorest(config: dict) -> List[dict]:
...
+    repo_url: str = config['repoHttpsUrl']
+
+    tsp_projects = config['relatedTypeSpecProjectFolder']
+    if isinstance(tsp_projects, str):
+        tsp_projects = [tsp_projects]
+
+    # mgmt tsp project folder follow the pattern, e.g. specification/deviceregistry/DeviceRegistry.Management
+    arm = any(re.match(r'specification[\\/](.*)[\\/](.*)[\\.]Management', tsp_project) for tsp_project in tsp_projects)
+    if not arm:
+        return sdk_automation_typespec_data(config)
```
Comment (XiaofeiCao) [inline eng/mgmt/automation/generate.py:229]: main change here, others are duplicating `sdk_automation_typespec` from `generate_data.py`.

File: eng/mgmt/automation/generate.py [hunk 2]
```diff
@@ -199,6 +207,116 @@ def sdk_automation_autorest(config: dict) -> List[dict]:
...
+                require_sdk_integration = git_pom_item.startswith('A ')
+
+            # parse service and module
+            match = re.match(r'sdk[\\/](.*)[\\/](.*)', sdk_folder)
+            service = match.group(1)
+            module = match.group(2)
+
+            # TODO (weidxu): move to typespec-java
+            if require_sdk_integration:
+                set_or_default_version(sdk_root, GROUP_ID, module)
```
Comment (XiaofeiCao) [inline eng/mgmt/automation/generate.py:275]: previous failure is due to `GROUP_ID` not correct, `com.azure.resourcemanager` vs `com.azure`.

File: eng/mgmt/automation/generate.py [hunk 3]
```diff
@@ -199,6 +207,116 @@ def sdk_automation_autorest(config: dict) -> List[dict]:
...
+            logging.info('Command line: ' + ' '.join(cmd))
+            output = subprocess.check_output(cmd, cwd=sdk_root)
+            output_str = str(output, 'utf-8')
+            script_return = output_str.splitlines()[-1] # the path to sdk folder
+            sdk_folder = os.path.relpath(script_return, sdk_root)
+            logging.info('SDK folder: ' + sdk_folder)
+            if sdk_folder:
+                succeeded = True
+        except subprocess.CalledProcessError as error:
+            logging.error(f'TypeSpec-Project-Process.ps1 fail: {error}')
```
Comment (weidongxu-microsoft) [inline eng/mgmt/automation/generate.py:251]: Would be using tsp-client 0.7.0 for this part.

Will put to util py for sharing.

File: eng/mgmt/automation/generate.py [hunk 4]
```diff
@@ -199,6 +207,117 @@ def sdk_automation_autorest(config: dict) -> List[dict]:
...
+    tsp_projects = config['relatedTypeSpecProjectFolder']
+    if isinstance(tsp_projects, str):
+        tsp_projects = [tsp_projects]
+
+    for tsp_project in tsp_projects:
+        # mgmt tsp project folder follow the pattern, e.g. specification/deviceregistry/DeviceRegistry.Management
+        if re.match(r'specification[\\/](.*)[\\/](.*)[\\.]Management', tsp_project):
+            packages.append(sdk_automation_typespec_project(tsp_project, config))
+        else:
+            packages.append(sdk_automation_typespec_project_data(tsp_project, config))
```
Comment (XiaofeiCao) [inline eng/mgmt/automation/generate.py:225]: support 
```json
"relatedTypeSpecProjectFolder": [
    "specification/deviceregistry/DeviceRegistry.Management",
    "specification/cognitiveservices/ContentSafety"
]
```

though in real life, automation pipeline passed in two separate `generateInput.json` file...

Comment (weidongxu-microsoft) [reply eng/mgmt/automation/generate.py:225]: I don't think this is what we need to support.

The configure is once invocation per project.

If these is multiple project, the output files would be mixed and probably not able to do a clean PR (client_version etc.)

File: eng/mgmt/automation/generate.py [hunk 5]
```diff
@@ -199,6 +207,117 @@ def sdk_automation_autorest(config: dict) -> List[dict]:
...
+        logging.info('Command line: ' + ' '.join(cmd))
+        output = subprocess.check_output(cmd, cwd=sdk_root)
+        output_str = str(output, 'utf-8')
+        script_return = output_str.splitlines()[-1] # the path to sdk folder
+        sdk_folder = os.path.relpath(script_return, sdk_root)
+        logging.info('SDK folder: ' + sdk_folder)
+        if sdk_folder:
+            succeeded = True
+    except subprocess.CalledProcessError as error:
+        logging.error(f'TypeSpec-Project-Process.ps1 fail: {error}')
```
Comment (weidongxu-microsoft) [inline eng/mgmt/automation/generate.py:256]: can you extract this part?

File: eng/mgmt/automation/generate.py [hunk 6]
```diff
@@ -199,6 +207,117 @@ def sdk_automation_autorest(config: dict) -> List[dict]:
...
+        check_call(cmd, sdk_root)
+        cmd = ['git', 'status', '--porcelain', os.path.join(sdk_folder, 'pom.xml')]
+        logging.info('Command line: ' + ' '.join(cmd))
+        output = subprocess.check_output(cmd, cwd=sdk_root)
+        output_str = str(output, 'utf-8')
+        git_items = output_str.splitlines()
+        if len(git_items) > 0:
+            git_pom_item = git_items[0]
+            # new pom.xml implies new SDK
+            require_sdk_integration = git_pom_item.startswith('A ')
```
Comment (weidongxu-microsoft) [inline eng/mgmt/automation/generate.py:271]: and this part

File: eng/mgmt/automation/generate.py (hunk 5 above)
Comment (XiaofeiCao) [reply eng/mgmt/automation/generate.py:256]: sure

Comment (XiaofeiCao) [reply eng/mgmt/automation/generate.py]: extracted `generate_typespec_project`

**Review by weidongxu-microsoft — APPROVED** [review]

**Pull request merged**